File: LNPG_08_TAD/c/book.c
#include <stdio.h>
#include <string.h>

typedef struct
{
    char title[50]; 
    char author[20];
    int publishedYear;
    int availableCopies;
} Book;

void initialize(Book *book, char title[50], char author[20], int publishedYear, int availableCopies)
{
    strncpy(book->title, title, sizeof(book->title) - 1);
    book->title[sizeof(book->title) - 1] = '\0'; 
    strncpy(book->author, author, sizeof(book->author) - 1);
    book->author[sizeof(book->author) - 1] = '\0'; 
    book->publishedYear = publishedYear;
    book->availableCopies = availableCopies;
}

int toLend(Book *book)
{
    if (book->availableCopies > 0)
    {
        book->availableCopies--;
        return 1;
    }
    return 0;
}

char *getTitle(Book *book)
{
    return book->title;
}

int main()
{
    Book book1;
    initialize(&book1, "A volta dos que não foram", "Jorge", 1999, 10);
    printf("%s\n", getTitle(&book1)); // Corrected printf statement
    return 0;
}

File: LNPG_08_TAD/java/Book.java
package LNPG_08_TAD.java;

public class Book {
    private String title;
    private String author;
    private int publishedYear = 0;
    private int availableCopys = 0;

    public Book(String title, String author) {
        this.title = title;
        this.author = author;
    }
    public Book(String title, String author, int publishedYear, int availableCopys) {
        this.title = title;
        this.author = author;
        this.publishedYear = publishedYear;
        this.availableCopys = availableCopys;
    }


    public String getTitle() {
        return this.title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getAuthor() {
        return this.author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public int getPublishedYear() {
        return this.publishedYear;
    }

    public void setPublishedYear(int publishedYear) {
        this.publishedYear = publishedYear;
    }

    public int getAvailableCopys() {
        return this.availableCopys;
    }

    public void setAvailableCopys(int availableCopys) {
        this.availableCopys = availableCopys;
    }

    public void toLend() throws Exception {
        if(this.availableCopys > 0) {
            this.availableCopys -= 1;
            return;
        }
        throw new Exception("Doesn't have available copys");
    }

    public void returnCopy() {
        this.availableCopys += 1;
    }

    public boolean hasAvailable() {
        return this.availableCopys > 0;
    }

    public String getInfo() {
        return String.format("Title: %s\nAuthor: %s\nPublished year: %d\nAvailable copys: %d", this.title, this.author, this.publishedYear, this.availableCopys);
    }

    @Override
    public String toString() {
        return this.getInfo();
    }
}